src/meta/processors/usersMan/AuthenticationProcessor.cpp

```diff
@@ -0,0 +1,271 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/usersMan/AuthenticationProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void CreateUserProcessor::process(const cpp2::CreateUserReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    const auto& user = req.get_user();
+    auto ret = getUserId(user.get_account());
+    if (ret.ok()) {
+        if (req.get_missing_ok()) {
+            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+        } else {
+            LOG(ERROR) << "Create User Failed :" << user.get_account() << " have existed";
+            resp_.set_code(cpp2::ErrorCode::E_EXISTED);
+        }
+        resp_.set_id(to(ret.value(), EntryType::USER));
+        onFinished();
+        return;
+    }
+
+    UserID userId = autoIncrementId();
+    std::vector<kvstore::KV> data;
+    data.emplace_back(MetaServiceUtils::indexUserKey(user.get_account()),
+                      std::string(reinterpret_cast<const char*>(&userId), sizeof(userId)));
+    LOG(INFO) << "Create User " << user.get_account() << ", userId " << userId;
+    data.emplace_back(MetaServiceUtils::userKey(userId),
+                      MetaServiceUtils::userVal(req.get_encoded_pwd(), user));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    resp_.set_id(to(userId, EntryType::USER));
+    doPut(std::move(data));
+}
+
+
+void AlterUserProcessor::process(const cpp2::AlterUserReq& req) {
```

Comment:
It seems the alter is to change the whole value. 
I want to know what's the language for alter user?

Reply:
No, Only change properties that require alter.

**Alter user sentences:**
```
ALTER USER account WITH with_user_opt_list

with_user_opt_item
    : KW_ACCOUNT KW_LOCK
    | KW_ACCOUNT KW_UNLOCK
    | KW_MAX_QUERIES_PER_HOUR INTEGER
    | KW_MAX_UPDATES_PER_HOUR INTEGER
    | KW_MAX_CONNECTIONS_PER_HOUR INTEGER
    | KW_MAX_USER_CONNECTIONS INTEGER
    ;

with_user_opt_list
    : with_user_opt_item
    | with_user_opt_list COMMA with_user_opt_item
    ;
```